### PR TITLE
Support interrupting a deploy job

### PR DESCRIPTION
**Pull request by olliewalsh**
The tripleo-deploy.sh oc exec shell creates a new process group. Can use
this to terminate the deployment.

Set terminateGracePeriod to 60s for the deploy job container.
Add a signal handler goroutine to the deploy job that exec a new
tripleo-deploy-term.sh script on the openstackclient.
This script will send a SIGTERM to the process group and, if it has not
not terminated after ~30s~50s, send a SIGKILL.

### CODE REVIEW

File: templates/openstackclient/bin/tripleo-deploy.sh [hunk 1]
```diff
@@ -4,6 +4,18 @@ umask 022
 
 export PWD=/home/cloud-admin
 CONFIG_VERSION=${CONFIG_VERSION:?"Please set CONFIG_VERSION."}
+sudo bash -c 'mkdir -p /var/run/tripleo-deploy && chown '$(whoami)' /var/run/tripleo-deploy'
```
Comment (stuggi) [inline templates/openstackclient/bin/tripleo-deploy.sh:7]: could also use an emptyVol for /var/run/tripleo-deploy and change the owner in the init container, but I am also ok to have it as is

Comment (dprince) [reply templates/openstackclient/bin/tripleo-deploy.sh:7]: wait, if we use emptyDir for /var/run/tripleo-deploy won't that potentially get rid of the ansible log too?

Comment (dprince) [reply templates/openstackclient/bin/tripleo-deploy.sh:7]: wait, if we use emptyDir for /var/run/tripleo-deploy won't that potentially get rid of the ansible log too?

Comment (dprince) [reply templates/openstackclient/bin/tripleo-deploy.sh:7]: Oh, Nevermind. This is just the RUNDIR. Not workdir.

Comment (dprince) [reply templates/openstackclient/bin/tripleo-deploy.sh:7]: Oh, Nevermind. This is just the RUNDIR. Not workdir.

**Comment by dprince** [comment]
Looks good to me. The /var/run/tripleo-deploy emptydir could come in a later patch

**Comment by dprince** [comment]
Looks good to me. The /var/run/tripleo-deploy emptydir could come in a later patch

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**